src/module_PI/Raspberry_PI/main/EmailNotifier.java

```diff
@@ -0,0 +1,239 @@
+package module_PI.Raspberry_PI.main;
+
+import java.io.*;
+import java.util.*;
+
+import javax.activation.*;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+public class EmailNotifier {
+
+	private String gmailUsername;
+	private String gmailPassword;
+
+	private String recipientEmailAddress;
+	private String emailSubject = "Calico Home Security";
+	private String emailMessage;
+	private String attachmentFilePath;
+
+	private Properties mailSessionProperties;
+	private Session javaMailSession;
+	private Message javaMailMessage;
+	private BodyPart messageBodyPart;
+	private Multipart multipart;
+	private DataSource attachmentDataSource;
+
+	public EmailNotifier(String recipientEmailAddress, String message) {
+		this(recipientEmailAddress, message, null);
+	}
+
+	public EmailNotifier(String recipientEmailAddress, String message, String pathToAttachment) {
+		this.loadGmailCredentials();
+		this.setRecipientEmailAddress(recipientEmailAddress);
+		this.emailMessage = message;
+		this.setAttachmentFilePath(pathToAttachment);
+		buildPropertiesAndCreateSession();
+	}
+
+	private void loadGmailCredentials() {
+		try {
+			Properties gmailProperties = new Properties();
+			FileInputStream in = new FileInputStream("gmailCredentials");
```

Comment:
Sorry - I should have used a different example. That one showing it writing to a file was more for an example rather than something you have to do. That only confused things... This example is better at illustrating what I was meaning: https://www.tutorialspoint.com/java/java_properties_class.htm 

You would define the Properties object as a Member variable in the class, not inside a specific method. Right now you are defining it inside the `parseGmailCredentials()` method. Thats why you aren't able to access them elsewhere. You only want to do the `put` calls inside one of the methods.

So instead of what you have, you'll want to define it as part of PiApp class:

```java
public class PiApp {

    public static Properties GMAIL_CREDENTIALS = new Properties(); // <<<<<<<<<<<<<<

    /**
     * @param args Gmail Username and Password for sending mail
     */
    public static void main(String[] args) {
        parseArgs(args);
        ......
    }

    private static void parseArgs(String[] args) {
        GMAIL_CREDENTIALS.put("gmailUsername", args[0]);
        GMAIL_CREDENTIALS.put("gmailPassword", args[1]);
    }

......
```
(_Note the change in name since it is a static member variable and we want to follow standard naming conventions._)

Now anywhere else outside this class you should be able to access it via something similar to `PiApp.GMAIL_CREDENTIALS.getProperty(...);`

Doing it this way you'll never need to access any file. Does that make more sense?